File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> ans;
        map<int,int> mp;
        queue<TreeNode*> q ;
        q.push(root);
        int lvl = 0;
        q.push(NULL);
        while(!q.empty() && q.front()!=NULL)
        {
          TreeNode* front = q.front();
          q.pop();
          if(front->left)
             q.push(front->left);
          if(front->right)
             q.push(front->right);
          mp[lvl] = front->val;
          if(q.front()==NULL)
          {
             lvl++;
             q.pop();
             q.push(NULL);
          }     
        } 
        for(auto it : mp)
        {
            ans.push_back(it.second);
        }
        return ans;
    }
};

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        def helper(root):
            if not root:
                return []
            res = []    
            q = deque()
            q.append(root)
            while q :
                rtnode = None
                l = len(q)
                for i in range(l):
                    node = q.popleft() 
                    if node :
                        rtnode = node
                        q.append(node.left)
                        q.append(node.right)
                if rtnode:        
                    res.append(rtnode.val)   
            return res
        return helper(root)        




        